src/plonk/lookup/prover.rs

```diff
@@ -72,12 +69,12 @@ impl<F: FieldExt> Argument<F> {
     /// - constructs Permuted<C> struct using permuted_input_value = A', and
     ///   permuted_table_expression = S'.
     /// The Permuted<C> struct is used to update the Lookup, and is then returned.
-    pub(in crate::plonk) fn commit_permuted<'a, C, T: TranscriptWrite<C>>(
+    pub(in crate::plonk) fn commit_permuted<'a, C, S: ChallengeSpace<C>, T: TranscriptWrite<C, S>>(
         &self,
         pk: &ProvingKey<C>,
         params: &Params<C>,
         domain: &EvaluationDomain<C::Scalar>,
-        theta: ChallengeTheta<C>,
+        theta: ChallengeScalar<C>,
```

Comment:
This could be `ChallengeScalar<C, ChallengeScalarType::Theta>` where `ChallengeScalarType::Theta` is a marker type.

Reply:
You're right, it was unnecessary to introduce a `ChallengeScalarType` enum. I've force-pushed to undo that change and preserve the old type distinction.